ArcGISRuntimeSDKQt_CppSamples/EditData/EditFeatureAttachments/EditFeatureAttachments.cpp

```diff
@@ -186,8 +193,10 @@ void EditFeatureAttachments::deleteAttachment(int index)
       if (m_selectedFeature->loadStatus() == LoadStatus::Loaded)
       {
         disconnect(m_attachmentConnection);
-        auto future = m_selectedFeature->attachments()->deleteAttachmentAsync(index);
-        Q_UNUSED(future)
+        m_attachmentListModel->deleteAttachmentAsync(index).then(this, [this]()
```

Comment:
Please audit the code paths. I see now that with my recommendation to store the AttachmentListModel as a class member, there is risk of timing problems where it could be the attachment list model for a different feature if we're not careful about when we assign it.

Give it some thought. If my suggestion makes the code harder to maintain, I'm ok if you change it back. Now that you've addressed any potential confusion about what happens when you call `m_selectedFeature->attachments(false, false)` or with different bools at different times, I think the my original concern is minimized.

Reply:
Changed all to `m_selectedFeature->attachments(false, false)` pattern.